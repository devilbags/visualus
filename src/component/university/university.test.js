import React from 'react';
import { shallow, mount } from 'enzyme';
import University from './university';

const setup = (setupProps = {}) => {
  const defaultProps = {
    loading: false,
    valueList: [
      {
        id: 1,
        name: 'Boston University',
        alias: 'BU'
      }
    ],
    onInputValueChange: jest.fn()
  };
  const props = { ...defaultProps, ...setupProps };
  const wrapper = shallow(
    <University
      loading={props.loading}
      valueList={props.valueList}
      onInputValueChange={props.onInputValueChange}
    />
  );
  const enzymeWrapper = mount(<University {...props} />);
  return {
    props,
    wrapper,
    enzymeWrapper
  };
};

describe('University component', () => {
  it('Should render without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('Should fire onInputValueChange when entered search value', () => {
    const { enzymeWrapper, props } = setup();
    const input = enzymeWrapper.find('Search');
    expect(props.onInputValueChange).not.toHaveBeenCalled();
    input.props().onSearch('test');
    expect(props.onInputValueChange).toHaveBeenCalledTimes(1);
  });
});
