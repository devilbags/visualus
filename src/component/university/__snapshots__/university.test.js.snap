// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`University component Should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <University
    loading={false}
    onInputValueChange={[Function]}
    valueList={
        Array [
            Object {
              "alias": "BU",
              "id": 1,
              "name": "Boston University",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Row
          gutter={0}
>
          <Col
                    lg={
                              Object {
                                        "offset": 4,
                                        "span": 16,
                                      }
                    }
                    md={
                              Object {
                                        "offset": 3,
                                        "span": 18,
                                      }
                    }
                    sm={
                              Object {
                                        "offset": 0,
                                        "span": 24,
                                      }
                    }
                    xl={
                              Object {
                                        "offset": 5,
                                        "span": 14,
                                      }
                    }
          >
                    <Search
                              enterButton={true}
                              inputPrefixCls="ant-input"
                              onSearch={[Function]}
                              placeholder="Search by name. e.g. Boston University"
                              prefixCls="ant-input-search"
                    />
          </Col>
</Row>,
        <List
          bordered={false}
          dataSource={
                    Array [
                              Object {
                                "alias": "BU",
                                "id": 1,
                                "name": "Boston University",
                              },
                            ]
          }
          itemLayout="vertical"
          loading={false}
          pagination={false}
          prefixCls="ant-list"
          renderItem={[Function]}
          size="large"
          split={true}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Col
            lg={
                        Object {
                                    "offset": 4,
                                    "span": 16,
                                  }
            }
            md={
                        Object {
                                    "offset": 3,
                                    "span": 18,
                                  }
            }
            sm={
                        Object {
                                    "offset": 0,
                                    "span": 24,
                                  }
            }
            xl={
                        Object {
                                    "offset": 5,
                                    "span": 14,
                                  }
            }
>
            <Search
                        enterButton={true}
                        inputPrefixCls="ant-input"
                        onSearch={[Function]}
                        placeholder="Search by name. e.g. Boston University"
                        prefixCls="ant-input-search"
            />
</Col>,
          "gutter": 0,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Search
              enterButton={true}
              inputPrefixCls="ant-input"
              onSearch={[Function]}
              placeholder="Search by name. e.g. Boston University"
              prefixCls="ant-input-search"
/>,
            "lg": Object {
              "offset": 4,
              "span": 16,
            },
            "md": Object {
              "offset": 3,
              "span": 18,
            },
            "sm": Object {
              "offset": 0,
              "span": 24,
            },
            "xl": Object {
              "offset": 5,
              "span": 14,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "enterButton": true,
              "inputPrefixCls": "ant-input",
              "onSearch": [Function],
              "placeholder": "Search by name. e.g. Boston University",
              "prefixCls": "ant-input-search",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bordered": false,
          "dataSource": Array [
            Object {
              "alias": "BU",
              "id": 1,
              "name": "Boston University",
            },
          ],
          "itemLayout": "vertical",
          "loading": false,
          "pagination": false,
          "prefixCls": "ant-list",
          "renderItem": [Function],
          "size": "large",
          "split": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Row
            gutter={0}
>
            <Col
                        lg={
                                    Object {
                                                "offset": 4,
                                                "span": 16,
                                              }
                        }
                        md={
                                    Object {
                                                "offset": 3,
                                                "span": 18,
                                              }
                        }
                        sm={
                                    Object {
                                                "offset": 0,
                                                "span": 24,
                                              }
                        }
                        xl={
                                    Object {
                                                "offset": 5,
                                                "span": 14,
                                              }
                        }
            >
                        <Search
                                    enterButton={true}
                                    inputPrefixCls="ant-input"
                                    onSearch={[Function]}
                                    placeholder="Search by name. e.g. Boston University"
                                    prefixCls="ant-input-search"
                        />
            </Col>
</Row>,
          <List
            bordered={false}
            dataSource={
                        Array [
                                    Object {
                                      "alias": "BU",
                                      "id": 1,
                                      "name": "Boston University",
                                    },
                                  ]
            }
            itemLayout="vertical"
            loading={false}
            pagination={false}
            prefixCls="ant-list"
            renderItem={[Function]}
            size="large"
            split={true}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Col
              lg={
                            Object {
                                          "offset": 4,
                                          "span": 16,
                                        }
              }
              md={
                            Object {
                                          "offset": 3,
                                          "span": 18,
                                        }
              }
              sm={
                            Object {
                                          "offset": 0,
                                          "span": 24,
                                        }
              }
              xl={
                            Object {
                                          "offset": 5,
                                          "span": 14,
                                        }
              }
>
              <Search
                            enterButton={true}
                            inputPrefixCls="ant-input"
                            onSearch={[Function]}
                            placeholder="Search by name. e.g. Boston University"
                            prefixCls="ant-input-search"
              />
</Col>,
            "gutter": 0,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Search
                enterButton={true}
                inputPrefixCls="ant-input"
                onSearch={[Function]}
                placeholder="Search by name. e.g. Boston University"
                prefixCls="ant-input-search"
/>,
              "lg": Object {
                "offset": 4,
                "span": 16,
              },
              "md": Object {
                "offset": 3,
                "span": 18,
              },
              "sm": Object {
                "offset": 0,
                "span": 24,
              },
              "xl": Object {
                "offset": 5,
                "span": 14,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "enterButton": true,
                "inputPrefixCls": "ant-input",
                "onSearch": [Function],
                "placeholder": "Search by name. e.g. Boston University",
                "prefixCls": "ant-input-search",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bordered": false,
            "dataSource": Array [
              Object {
                "alias": "BU",
                "id": 1,
                "name": "Boston University",
              },
            ],
            "itemLayout": "vertical",
            "loading": false,
            "pagination": false,
            "prefixCls": "ant-list",
            "renderItem": [Function],
            "size": "large",
            "split": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
